dotCMS/src/main/java/com/dotmarketing/business/VersionableFactoryImpl.java

```diff
@@ -381,12 +393,27 @@ protected void deleteContentletVersionInfo(String id, long lang) throws DotDataE
 
         if(UtilMethods.isSet(contv.getIdentifier())) {
         	HibernateUtil.delete(contv);
-        	icache.removeContentletVersionInfoToCache(id, lang);
+        	this.icache.removeContentletVersionInfoToCache(id, lang);
         }
 	}
-	
-	
 
-	
-	
+	/**
+	 * Returns the list of dotCMS objects whose information can be looked up via
+	 * APIs before performing a database lookup. This approach allows the
+	 * information to be obtained by checking the cache first instead of going
+	 * directly to the database, which is more expensive.
+	 * 
+	 * @return The list of classes representing the objects that need to be
+	 *         looked up using their respective APIs before performing a
+	 *         database query.
+	 */
+	private List<Class<?>> getVersionableWhitelist() {
+		List<Class<?>> whitelist = new ArrayList<>();
```

Comment:
if you want you can use CollectionsUtils, such as list(Container.class, Template.class....);

Also it could be a java.util.Set, the contains will be faster than a java.util.List.